bpm/bonita-core/bonita-process-instance/bonita-process-instance-impl/src/main/java/org/bonitasoft/engine/core/process/instance/event/impl/EventInstanceServiceImpl.java

```diff
@@ -233,20 +229,13 @@ public void deleteWaitingEvent(final SWaitingEvent waitingEvent) throws SWaiting
 
     @Override
     public void deleteWaitingEvents(final SFlowNodeInstance flowNodeInstance) throws SWaitingEventModificationException, SBonitaReadException {
-        final OrderByOption orderByOption = new OrderByOption(SWaitingEvent.class, BuilderFactory.get(SWaitingMessageEventBuilderFactory.class)
-                .getFlowNodeNameKey(), OrderByType.ASC);
-        final FilterOption filterOption = new FilterOption(SWaitingEvent.class, BuilderFactory.get(SWaitingMessageEventBuilderFactory.class)
-                .getFlowNodeInstanceIdKey(), flowNodeInstance.getId());
-        final List<FilterOption> filters = Collections.singletonList(filterOption);
-        final QueryOptions queryOptions = new QueryOptions(0, 100, Collections.singletonList(orderByOption), filters, null);
-
-        List<SWaitingEvent> waitingEvents = searchWaitingEvents(SWaitingEvent.class, queryOptions);
+        List<SWaitingEvent> waitingEvents;
         do {
+            waitingEvents = getPersistenceService().selectList(new SelectListDescriptor<SWaitingEvent>("getWaitingEventsOfFlowNode", Collections.<String, Object>singletonMap("flowNodeInstanceId", flowNodeInstance.getId()), SWaitingEvent.class, new QueryOptions(0, 100)));

```

Comment:
It's better to use BuilderFactory.get(SWaitingMessageEventBuilderFactory.class).getFlowNodeInstanceIdKey() to retrieve the column name.